src/VisualStudio/Core/Def/Options/ExportVisualStudioStorageReadFallbackAttribute.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.VisualStudio.LanguageServices.Options;
+
+/// <summary>
+/// Use this attribute to declare a custom fallback reader for an option stored in Visual Studio storage.
+/// </summary>
+[MetadataAttribute]
+[AttributeUsage(AttributeTargets.Class)]
+internal sealed class ExportVisualStudioStorageReadFallbackAttribute : ExportAttribute
+{
+    /// <summary>
+    /// Option unique name. <see cref="OptionDefinition.ConfigName"/>.
```

Comment:
Is this always `.editorconfig` key?

Reply:
Yes, for options stored in .editorconfig.